### DISCUSSION TITLE
500 error on signup even with SECURITY DEFINER trigger and function

### DISCUSSION THREAD

**Discussion by kaelaworthen** [supabase/supabase#23555] [Questions]
I have a function that is triggered whenever a user is created in the auth.users table. The function has a SECURITY DEFINER attribute on it and successfully does the following:
* creates a row in the estates table
* creates two different rows in the profiles table with the estate id
* creates all the default tasks in the task_assignments table with the estate id

However, when I try to get it to also create some default rows in the financial_accounts table, it doesn't work and instead causes the signup to give a 500 error. I've tried:
* turning off row-level security for the financial_accounts table (did not work)
* running the part of the funciton to create the financial_accounts separately as a manual sql command (does work on its own, so not an issue with the structure of the code)
* commenting out the part of the code that creates the financial accounts, which then allows everything to work correctly, but obviously does not create the financial accounts needed
* adding an EXCEPTION clause on the financial_accounts creation, which allows the signup to work again and everything else to be created correctly, but still does not create the financial_accounts rows

Any suggestions on how I can address this?

Here are the relevant pieces of the function:

```
CREATE OR REPLACE FUNCTION public.auth_new_signup() 
RETURNS TRIGGER AS $$
DECLARE
  {{variables used in creating new estate and tasks}}
BEGIN
  {{code for creating new estate, profiles, and tasks}}

  -- Attempt to insert into financial_accounts
  BEGIN
    INSERT INTO financial_accounts ({{fields being inserted}})
    VALUES ({{values being inserted}});

    INSERT INTO financial_accounts ({{fields being inserted}})
    VALUES ({{values being inserted}});

  EXCEPTION
    WHEN OTHERS THEN
      RAISE NOTICE 'Failed to insert into financial_accounts: %', SQLERRM;
  END;

  RETURN NEW;
END;
$$ LANGUAGE plpgsql SECURITY DEFINER;
```

Trigger looks like this:
```
create trigger auth_new_signup
  after insert on auth.users
  for each row execute procedure public.auth_new_signup();
```

Again, the RLS for the financial_accounts table does not seem to be the issue since even turning off RLS entirely for that table didn't work, but if it's helpful, I have it defined as anyone can select, insert, or update rows in the financial_accounts table if the estate_id on that table is the same as the estate_id on their profile

Thanks so much for any assistance you can provide!

**Comment by GaryAustin1** [comment]
Please look in the dashboard Postgres logs for what the error is.

**Reply by GaryAustin1** [reply]
My guess is probably you don't have `public.financial_accounts` or set the search path to include public.
